development/git_lfs.rst

```diff
@@ -0,0 +1,156 @@
+
+   Username for 'https://s3.lsst.codes': <Empty>
+   Password for 'https://s3.lsst.codes': <Empty>
+
+There is no username or password for LSST's S3 service.
+
+Once your credentials are cached, you won't need to repeat this process on your system (:ref:`unless you opted for the cache-based credential helper <>`).
+
+.. _git-lfs-using:
+
+Using Git LFS-enabled repositories
+==================================
+
+Git LFS operates transparently to the user.
+*Just use the repo as you normally would any other Git repo.*
+All of the regular Git commands just work, whether you are working with LFS-managed files or not.
+
+There are two caveats for working with LFS: use of HTTPS is required, and Git LFS must be told to track new binary file types.
+
+DM's LFS implementation mandates the HTTPS transport protocol, not SSH when working with GitHub remotes.
+Developers used to working with `ssh-agent <http://www.openbsd.org/cgi-bin/man.cgi?query=ssh-agent&sektion=1>`_ for passwordless GitHub interaction should use a :ref:`Git credential helper <git-credential-helper>`, and follow the directions above for configuring their credentials.
+
+In an LFS-backed repository, you need to specify what files are stored by LFS rather than regular Git storage.
+You can run
+
+.. code-block:: bash
+
+   git lfs track
+
+to see what file types are being tracked by LFS in your repository.
+:ref:`We describe how to track additional file types below <git-lfs-tracking>`.
+
+.. _git-lfs-tracking:
+
+Tracking new file types
+=======================
+
+Only file types that are specifically *tracked* are stored in Git LFS rather than the standard Git storage.
+
+To see what file types are already being tracked in a repository:
+
+.. code-block:: bash
+
+   git lfs track
+
+To track a *new* file type (FITS files, for example):
+
+.. code-block:: bash
+
+   git lfs track "*.fits"
+
+Git LFS stores information about tracked types in the :file:`.gitattributes` file.
+This file is part of the repo and tracked by Git itself.
+
+You can ``git add``, ``commit`` and do any other Git operations against these Git LFS-managed files.
+
+To see what files are being managed by Git LFS, run:
+
+.. code-block:: bash
+
+   git lfs ls-files
+
+.. _git-lfs-create:
+
+Creating a new Git LFS-enabled repository
+=========================================
+
+Configuring a new Git repository to store files with DM's Git LFS is easy.
+First, initialize the current directory as a repository:
+
+.. code-block:: bash
+
+   git init .
+
+Make a file called :file:`.gitconfig` *within the repository*, and write these lines into it:
+
+.. code-block:: text
+
+   [lfs]
+        url = https://git-lfs.lsst.codes
+        batch = false
+
+Next, track some files types.
+For example, to have FITS and ``*.gz`` files tracked by Git LFS,
+
+.. code-block:: bash
+
+   git lfs track "*.fits"
+   git lfs track "*.gz"
+
+Add and commit the :file:`.gitconfig` and :file:`.gitattributes` files to your repository.
```

Comment:
So given that this section is about making new LFS-based repos, I should probably just say `.lfsconfig` and ignore that `.gitconfig` was ever a thing for us. Or is it necessary to mention the deprecation process happening now?